utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -79,6 +79,43 @@ impl TrieValue for char {
     }
 }
 
+#[derive(PartialEq, Eq, Debug)]
+pub struct CodePointMapRange {
```

Comment:
I would much prefer if this were the more Rusty

```rust
pub struct CodePointMapRangeIterator<'a, T> {
    cpt: &'a CodePointTrie<T>,
    start: u32,
    end: u32,
    value: u32,
}

impl<'a, T> Iterator for CodePointMapRangeIterator<'a, T> {
    type Item = Range<u32>; // or (u32, u32) or similar
    fn next(...) -> ... { ... }
}

impl<T> CodePointTrie<T> {
    pub fn iter_ranges(&self) -> CodePointMapRangeIterator<T> { ... }
}
```

Reply:
It is useful to have both
- get_range() returning a Range (no need for a reference to the trie), as a power user API; especially when the ValueFilter parameter is added, but also to allow iteration over small parts of the Unicode range
- a Rust-y-idiomatic iterator (I guess like you are sketching out here) as a convenience API